sensorium/training/trainers.py

```diff
@@ -168,6 +252,40 @@ def full_objective(model, dataloader, data_key, *args, **kwargs):
         lr_decay_steps=lr_decay_steps,
     ):
 
+        if include_kldivergence and epoch == dec_starting_epoch:
+            # TODO: include hidden dimension
+            kmeans = KMeans(
+                n_clusters=cluster_number, n_init=kmeans_init, random_state=seed
+            )
+            model.train()
```

Comment:
I think this line should be after the things inside this `if` statement
like 
```
if include_kldivergence and epoch == dec_starting_epoch:
    kmeans = KMeans(
                n_clusters=cluster_number, n_init=kmeans_init, random_state=seed
            )
    ...
model.train()
if verbose and tracker is not None:
            print("=======================================")
```
Please correct me if I am wrong

Reply:
True